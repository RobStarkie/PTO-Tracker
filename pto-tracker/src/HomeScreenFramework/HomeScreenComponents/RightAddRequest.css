.rightComponentContainer{
    height: 650px;
    width: 100%;
    background-color: rgba(255,255,255,0.13);
    border-radius: 10px;
    backdrop-filter: blur(10px);
    border: 2px solid rgba(255,255,255,0.1);
    box-shadow: 0 15px 40px rgba(0,0,0,0.12);
    padding: 50px 35px;
    min-width: 400px;
}

.rightComponentForm *{
    font-family: 'Poppins',sans-serif;
    color: #000000;
    letter-spacing: 0.5px;
    outline: none;
    border: none;
  }

  .rightComponentForm button {
    width: 100%;
    margin-top: 40px;
    background-color: #000000;
    color: #ffffff;
    padding: 15px;
    font-size: 18px;
    font-weight: 600;
    border-radius: 5px;
    cursor: pointer;
    color: black;
    color: #FFF;
  }