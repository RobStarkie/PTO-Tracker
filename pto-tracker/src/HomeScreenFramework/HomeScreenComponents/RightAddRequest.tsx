import React, { useState } from "react";
import './RightAddRequest.css'

interface RightAddRequestProps {
    content: string;
}

const RightAddRequest: React.FC<RightAddRequestProps> = ({ content }) => {
    const [startDate, setStartDate] = useState<string | null>(null);
    const [endDate, setEndDate] = useState<string | null>(null);

    const handleNewHolidayRequest = () => {
        //submit new holiday request
        //submitNewRequest(startDate, endDate);
      };

    return (
        <div className="rightComponentContainer"> 
            <form className="rightComponentForm">
                <h1 className="rightComponentHeader">Enter a New Request</h1>
                <label htmlFor="startDate">Start Date</label>
                <input type="text" placeholder="dd/mm/yyyy" id="startDate" required onChange={e =>{
                    setStartDate(e.target.value);
                    e.preventDefault();
                }} />


                <label htmlFor="endDate">End Date</label>
                <input type="text" placeholder="dd/mm/yyyy" id="endDate" required onChange={e => {
                    setEndDate(e.target.value);
                    e.preventDefault();
                }}/>

                <button onClick={handleNewHolidayRequest}>Submit New Request</button>    
            </form>
        </div>
    );
};

export default RightAddRequest;