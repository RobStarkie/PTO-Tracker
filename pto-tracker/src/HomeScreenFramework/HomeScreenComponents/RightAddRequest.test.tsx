
import React from 'react';
import renderer from "react-test-renderer";
// import { render } from '@testing-library/react';
import RightAddRequest from './RightAddRequest';
import './RightAddRequest.css'
import { MemoryRouter } from "react-router-dom";
// import mockAxios from "jest-mock-axios";

// afterEach(() => {
//     mockAxios.reset();
// })

describe.only('RightAddRequest (Holiday Request Adder UI)', () => {
    it('renders correctly when loading', () => {

        const startDate = "2024-01-10";
        const endDate = "2024-01-15";
        const postcode = "CH5 3DR";

        const mockRequestData = {
            "StartDate": startDate,
            "EndDate": endDate,
            "Postcode": postcode
        };


        // Mock the props for this component:
        //  dummy string
        const mockContent = "Test Content";
        // mock function that returns a dummy token
        const mockGetToken = jest.fn(() => "test-token");
        // mock function that does nothing but can be used to simulate the function passed to the component
        const mockHandleAddNewHolidayRequest = jest.fn();

        // mockAxios.post.mockResolvedValueOnce({
        //     data: {
        //         "message": "New holiday request added successfully"
        //     },
        //     status: 201,
        //     statusText: 'Created',
        //     headers: {},
        //     config: {},
        // });

        
        const tree = renderer.create(
            <MemoryRouter>
                <RightAddRequest
                content={mockContent}
                getToken={mockGetToken}
                handleAddNewHolidayRequest={mockHandleAddNewHolidayRequest}
            />
        </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})