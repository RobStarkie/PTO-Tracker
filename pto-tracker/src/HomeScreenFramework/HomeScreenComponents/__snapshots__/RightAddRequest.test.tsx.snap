// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightAddRequest (Holiday Request Adder UI) renders correctly when loading 1`] = `
<div
  className="rightComponentContainer"
>
  <div
    className="rightComponentForm"
  >
    <h1
      className="rightComponentHeader"
    >
      Enter a New Request
    </h1>
    <label
      htmlFor="startDate"
    >
      Start Date
    </label>
    <input
      id="startDate"
      onChange={[Function]}
      placeholder="dd/mm/yyyy"
      required={true}
      type="date"
    />
    <label
      htmlFor="endDate"
    >
      End Date
    </label>
    <input
      id="endDate"
      onChange={[Function]}
      placeholder="dd/mm/yyyy"
      required={true}
      type="date"
    />
    <label
      htmlFor="postcode"
    >
      Hotel Location
    </label>
    <input
      id="postcode"
      onChange={[Function]}
      placeholder="ABC 123"
      required={true}
      type="string"
    />
    <button
      className="request-button"
      onClick={[Function]}
    >
      Submit New Request
    </button>
  </div>
</div>
`;
